projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/service/monitoring/DataJobMonitorTest.java

```diff
@@ -221,7 +221,7 @@ public void testWatchJobsWithTheSameStatus() throws IOException, ApiException {
         dataJobMonitor.watchJobs();
 
         var gauges = meterRegistry.find(DataJobMetrics.TAURUS_DATAJOB_TERMINATION_STATUS_METRIC_NAME).gauges();
-        Assertions.assertEquals(5, gauges.size());
+        Assertions.assertEquals(4, gauges.size());
```

Comment:
I don't understand this line, why is it changed from 5 to 4?

Reply:
The `watchJob` is invoked for this job while its termination status and execution id are the same as those persisted in the database. As a result, its termination status metrics are not updated (due to the fix) and no gauge is created.